### DISCUSSION TITLE
Ability to access log information in Edge Function context

### DISCUSSION THREAD

**Discussion by ChuckJonas** [supabase/supabase#17660] [Feature Requests]
One key observability feature missing is access to [logs](https://supabase.com/docs/guides/platform/logs?product=functions&source=function_edge_logs#log-source-reference) in the context of the function.  At a minimum, there should be some way to get the log id of the current execution. 

A more complete solution would allow for setting custom attributes from the request.  This would allow better support for [distributed tracing]((like https://www.w3.org/TR/trace-context/).

**Comment by laktek** [comment]
@ChuckJonas There should be an `execution_id` in logs metadata for Edge Functions, which should allow you to filter events from a single execution.

We also do have plans to implement tracing contexts. Currently I don't have an ETA for it.

**Comment by ChuckJonas** [comment]
@laktek sure, but is there some way to access the `execution_id` in the function runtime? 

```
serve(async (req) => {
 // how would I get `execution_id` here?
});
```

Since supabase functions seem to be framework-agnostic (no custom "context"), maybe the API gateway could inject a request header with this value?

**Comment by rookledookle** [comment]
Am looking for the same thing as @ChuckJonas ! Any updates?

**Comment by rookledookle** [comment]
Bump!
